pvlib/ivtools.py

```diff
@@ -262,6 +268,173 @@ def fit_sde_sandia(voltage, current, v_oc=None, i_sc=None, v_mp_i_mp=None,
+    elif celltype.lower() in ['gaas']:
+        C = 0.0003174  # valid for gallium arsenide
+        Eg = 2.163e-19  # in J, valid for gallium arsenide
+    elif celltype.lower() in ['cis', 'cigs', 'cdte', 'amorphous']:
+        raise NotImplementedError
+    else:
+        raise ValueError('Unknown cell type.')
+
+    # Conversion from %/K to A/K & V/K
+    alpha_sc = alpha_sc*i_sc/100
+    beta_voc = beta_voc*v_oc/100
+
+    def pv_fct(params, specs):
+        """Returns the system of equations used for computing the
+        single-diode 5 parameters.
+        To avoid the confusion in names with variables of container
+        function the '_' of the variables were removed.
+        """
+        # six input known variables
+        Isc = specs[0]
+        Voc = specs[1]
+        Imp = specs[2]
+        Vmp = specs[3]
+        betaoc = specs[4]
+        alphasc = specs[5]
+        # five parameters vector to find
+        IL = params[0]
+        Io = params[1]
+        a = params[2]
+        Rsh = params[3]
+        Rs = params[4]
+        # five equation vector
+        y = [0, 0, 0, 0, 0]
+
+        # 1st equation - short-circuit - eq(3) in [1]
+        y[0] = Isc - IL + Io*(np.exp(Isc*Rs/a) - 1.0) + Isc*Rs/Rsh
+        # 2nd equation - open-circuit Tref - eq(4) in [1]
+        y[1] = -IL + Io*(np.exp(Voc/a) - 1.0) + Voc/Rsh
+        # 3rd equation - Imp & Vmp - eq(5) in [1]
+        y[2] = Imp - IL + Io*(np.exp((Vmp+Imp*Rs)/a) - 1.0) + \
+            (Vmp+Imp*Rs)/Rsh
+        # 4th equation - Pmp derivated=0 -
+        # caution: eq(6) in [1] seems to be incorrect, take eq23.2.6 in [2]
+        y[3] = Imp - Vmp * ((Io/a)*np.exp((Vmp+Imp*Rs)/a) + 1.0/Rsh) / \
+            (1.0 + (Io*Rs/a)*np.exp((Vmp+Imp*Rs)/a) + Rs/Rsh)
+        # 5th equation - open-circuit T2 - eq (4) at temperature T2 in [1]
+        T2 = Tref + 2
+        Voc2 = (T2 - Tref)*betaoc + Voc  # eq (7) in [1]
+        a2 = a*T2/Tref  # eq (8) in [1]
+        IL2 = IL + alphasc*(T2-Tref)  # eq (11) in [1]
+        Eg2 = Eg*(1-C*(T2-Tref))  # eq (10) in [1]
+        Io2 = Io * (T2/Tref)**3 * np.exp(1/k * (Eg/Tref-Eg2/T2))  # eq (9)
+        y[4] = -IL2 + Io2*(np.exp(Voc2/a2) - 1.0) + Voc2/Rsh  # eq (4) at T2
+
+        return y
+
+    # initial guesses of variables for computing convergence:
+    # Values are taken from [2], p753
+    Rsh_i = 100.0
+    a_i = 1.5*k*Tref*cells_in_series/q
+    IL_i = i_sc
+    Io_i = i_sc * np.exp(-v_oc/a_i)
+    Rs_i = (a_i*np.log((IL_i-i_mp)/Io_i+1) - v_mp)/i_mp
+    # params_i : initial values vector
+    params_i = np.array([IL_i, Io_i, a_i, Rsh_i, Rs_i])
+
+    # specs of module
+    specs = np.array([i_sc, v_oc, i_mp, v_mp, beta_voc, alpha_sc])
+
+    # computing
+    sol = root(pv_fct, x0=params_i, args=specs, method='lm').x
+
+    # results
+    return {'I_L_ref': sol[0],
+            'I_o_ref': sol[1],
+            'a_ref': sol[2],
+            'R_sh_ref': sol[3],
+            'R_s': sol[4],
+            'alpha_sc': alpha_sc}
```

Comment:
My personal preference is to also record `EgRef`, `dEgdT`, `irrad_ref`, and `temp_ref` in the output, in order to fully capture the conditions of the fit. Ideally, these can then be passed straight through to `calcparams_desoto`, ensuring consistency.

`irrad_ref` should probably also be added as an input with a default value of `1000`. It's simply a pass through value, but I think it is an "implicit" part of the fit that the user should be aware of, and it corresponds to the default value in `calcparams_desoto`.